python/ray/data/tests/test_dataset.py

```diff
@@ -2554,6 +2554,14 @@ def test_groupby_map_groups_for_empty_dataset(ray_start_regular_shared):
     assert mapped.take_all() == []
 
 
+def test_groupby_map_groups_merging_empty_result(ray_start_regular_shared):
```

Comment:
Could we also add a test case for good error messages on returning None / incorrect data types?

Reply:
Added a logging in grouped_dataset.